lib/ansible/modules/clustering/consul_acl.py

```diff
@@ -245,16 +245,20 @@ def yml_to_rules(module, yml_rules):
                 rules.add_rule('event', Rule(rule['event'], rule['policy']))
             elif ('query' in rule and 'policy' in rule):
                 rules.add_rule('query', Rule(rule['query'], rule['policy']))
+            elif ('agent' in rule and 'policy' in rule):
```

Comment:
As more rule types are added, this structure is getting very repetitive. How about refactoring it into something like this:
```
def yml_to_rules(module, yml_rules):
    rules = Rules()
    if yml_rules:
        for rule in yml_rules:
            rule_added = False
            for rule_type in RULE_TYPES:
                if (rule_type in rule and 'policy' in rule):
                    rules.add_rule(rule_type, Rule(rule[rule_type], rule['policy']))
                    rule_added = True
                    break
            if not rule_added:
                module.fail_json(msg="a rule requires a key/service/event/query/agent/keyring/operator/session/node and a policy.")
    return rules
```